src/DurableTask.Core/TaskOrchestrationContext.cs

```diff
@@ -77,6 +77,11 @@ internal void ClearPendingActions()
         {
             object result = await ScheduleTaskInternal(name, version, taskList, typeof(TResult), parameters);
 
+            if (result == null && typeof(TResult).IsPrimitive)
+            {
+                throw new InvalidOperationException("Task result is null and cannot be cast to TResult. Please use the non-generic overload method.");
```

Comment:
Another option we could consider is this:
```suggestion
                return default(TResult);
```

This is what we did to handle a very similar situation in another part of the code: https://github.com/Azure/durabletask/blob/f2dc50f392ac03a5389d60f582184819310ffb8a/src/DurableTask.Core/Serializing/DataConverter.cs#L52-L61